File: src/ERP/acme.atena.domain/DTO/Inventory/SaidaProdutoEstoque.cs
using acme.atena.domain.DTO.Product;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace acme.atena.domain.DTO.Inventory
{
    public class SaidaProdutoEstoque : AbstractEntityCompetencia
    {
        public SaidaProdutoEstoque():base()
        {
            DataSaida = DateTime.Now;
        }
        public SaidaProdutoEstoque(Guid produtoId, Guid estoqueId, int quantidade) : this()
        {
            ProdutoId = produtoId;
            EstoqueId = estoqueId;
            Quantidade = quantidade;
        }

        public SaidaProdutoEstoque(Guid produtoId, Guid estoqueId, int quantidade, DateTime dataSaida) : this(produtoId, estoqueId, quantidade)
        {
            DataSaida = dataSaida;
        }

        public Guid ProdutoId { get; set; }
        public Guid EstoqueId { get; set; }
        public int Quantidade { get; set; }
        public DateTime DataSaida { get; set; }

        public virtual Estoque Estoque { get; set; }
        public virtual Produto Produto { get; set; }

    }
}

File: site/coreui-free-angular-admin-template-master/src/app/view-model/response/inventory/saida-produto-estoque.ts
import { Produto } from './../product/produto';
import { Estoque } from './estoque';
import { AbstractEntityCompetencia } from "../abstract-entity-competencia";

export class SaidaProdutoEstoque extends AbstractEntityCompetencia{
  public produtoId: string;
  public estoqueId: string;
  public quantidade: number;
  public dataSaida: Date;

  public  estoque: Estoque;
  public  produto: Produto;

}
